### PR TITLE
C#: Take more sources and sinks into account when reporting in the te…

### PR DESCRIPTION
**Pull request by michaelnebel**
…lemetry queries.

### CODE REVIEW

**Comment by michaelnebel** [comment]
DCA shows a performance decrease of around 3% and a significant DIL increase for the telemetry queries, which is not surprising as they now have a dependency to most of the `*Querl.qll` files.
DCA also reports a significant slowdown for some of the queries, but that could be caused by evaluation order. Will start another DCA experiment, where we just run a single telemetry query in isolation.

File: csharp/ql/lib/semmle/code/csharp/security/dataflow/flowsinks/AllSinks.qll [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  /**
+   * Add all models as data sinks.
+   */
+  private class SinkNodeExternal extends SinkNode {
+    SinkNodeExternal() { sinkNode(this, _) }
+  }
+}
+
+/**
```
Comment (hvitved) [inline csharp/ql/lib/semmle/code/csharp/security/dataflow/flowsinks/AllSinks.qll:81]: nit: I think this should be moved up before the `AllSinks` module.

File: csharp/ql/lib/semmle/code/csharp/security/dataflow/flowsources/AllSources.qll [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+   */
+  private class SourceNodeExternal extends SourceNode {
+    SourceNodeExternal() { sourceNode(this, _) }
+  }
+}
+
+/**
+ * A data flow source node.
+ */
+abstract class SourceNode extends DataFlow::Node { }
```
Comment (hvitved) [inline csharp/ql/lib/semmle/code/csharp/security/dataflow/flowsources/AllSources.qll:77]: same

**Review by hvitved — DISMISSED** [review]

**Review by hvitved — APPROVED** [review]

**Comment by michaelnebel** [comment]
DCA shows
- Increase in DIL size (expected).
- Increase in cache size (not unreasonable as we are adding lots of dependencies).
- A slowdown of the telemetry queries with around 15% (also not unexpected).
I also tried running the `cs/telemetry/supported-external-apis` locally from a cold cache on `mono` which confirms the (not unreasonable) slowdown.

**Pull request merged**